clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -272,18 +289,27 @@ void SYCL::fpga::BackendCompiler::ConstructJob(Compilation &C,
   assert((getToolChain().getTriple().getArch() == llvm::Triple::spir ||
           getToolChain().getTriple().getArch() == llvm::Triple::spir64) &&
          "Unsupported target");
-  assert((JA.getType() == types::TY_FPGA_AOCX ||
-          JA.getType() == types::TY_FPGA_AOCR) &&
-         "aoc type required");
+  StringRef ForeachExt = "aocx";
 
   ArgStringList CmdArgs{"-o",  Output.getFilename()};
-  for (const auto &II : Inputs) {
-    CmdArgs.push_back(II.getFilename());
-  }
+  ArgStringList ForeachArgs;
   CmdArgs.push_back("-sycl");
   if (Arg *A = Args.getLastArg(options::OPT_fsycl_link_EQ))
-    if (A->getValue() == StringRef("early"))
+    if (A->getValue() == StringRef("early")) {
       CmdArgs.push_back("-rtl");
+      ForeachExt = "aocr";
+    }
+
+  for (const auto &II : Inputs) {
+    if (II.getType() == types::TY_Tempfilelist) {
+      std::string FileName(II.getFilename());
```

Comment:
Suggest initializing before entering the condition and reusing [1]

Reply:
Okay.